buildSrc/src/main/java/org/elasticsearch/gradle/DependenciesGraphPlugin.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.tasks.TaskProvider;
+
+public class DependenciesGraphPlugin implements Plugin<Project> {
+
+    public void apply(Project project) {
+        final String url = System.getenv("SCA_URL");
+        final String token = System.getenv("SCA_TOKEN");
+        if (null == url || null == token) {
+            throw new GradleException("The environment variables SCA_URL and SCA_TOKEN need to be set before this task is run");
```

Comment:
I don't see us throwing from other plugins so I'm not sure if this is a good practice or not. I prefer this as it throws a meaningful error instead of the Task failing with "url is missing" and "token is missing"

Reply:
Ok, the subsequent failure made me realize there is a good reason why we don't throw in plugins :)